src/helpers/webp-support.js

```diff
@@ -0,0 +1,8 @@
+const canvasMock = typeof document === 'object' ? document.createElement('canvas') : {}
+canvasMock.width = canvasMock.height = 1
+
+// If canvas supports webp, the returned Data URLs would  be prefixed with "data:".
+// Thus data starts at index 5, after the prefix.
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs
+export const detectWebPSupport = () => canvasMock.toDataURL
+  ? canvasMock.toDataURL('image/webp').indexOf('image/webp') === 5 : false
```

Comment:
Thanks for putting this PR together, but I don't think something like this belongs in the design-system repo since it doesn't actually have anything to do with the CSS generated for the BackgroundImage component, but affects the `props.image` *value*.

This is a great example of where using the higher-order component (HOC) pattern would make a lot more sense and be more reusable without adding logic to a base-level style component. We could start another repo with utility HOCs if we end up making a lot of things like this, but I'd try to keep client-side logic out of this one.

An example HOC could be used like this:

```jsx
const WebPBackgroundImage = withWebP(BackgroundImage)

<WebPBackgroundImage
  fallback='path-to/fallback.jpg'
  image='path-to/image.webp'
/>
```

The HOC itself could look something like this:

```jsx
const withWebP = Component => {
  const WebPComponent = ({ image, fallback, ...rest }) => {
    const imageOrFallback = hasWebP ? image : fallback
    return <Component image={imageOrFallback} />
  }
  return WebPComponent
}
```

Reply:
Ok cool. Good to know the scope of WebP support can try out HOC pattern. Will close this PR. Thanks for your feedback.